File: src/main/java/me/tomasan7/jecnadesktop/parser/parsers/TimetableParser.java
package me.tomasan7.jecnadesktop.parser.parsers;

import me.tomasan7.jecnadesktop.data.Timetable;

/**
 * Is responsible for parsing any kind of formatted {@link String} (eg. JSON, XML) to {@link Timetable} instance.
 */
public interface TimetableParser
{
	Timetable parse (String source);
}

File: src/main/kotlin/me/tomasan7/jecnadesktop/parser/parsers/TimetableParser.kt
package me.tomasan7.jecnadesktop.parser.parsers

import me.tomasan7.jecnadesktop.data.Timetable

/**
 * Is responsible for parsing any kind of formatted [String] (eg. JSON, XML) to [Timetable] instance.
 */
interface TimetableParser
{
    /**
     * @throws me.tomasan7.jecnadesktop.parser.ParseException When the source isn't in correct format.
     */
    fun parse(source: String): Timetable
}